extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs

```diff
@@ -49,26 +59,53 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
 
             var invocationResult = context.GetInvocationResult();
 
-            if (invocationResult?.Value is IActionResult actionResult)
+            if (_httpResponseTypes.Any(type => invocationResult?.Value?.GetType() == type))
             {
-                ActionContext actionContext = new ActionContext(httpContext, httpContext.GetRouteData(), new ActionDescriptor());
-
-                await actionResult.ExecuteResultAsync(actionContext);
+                HandleHttpResult(invocationResult.Value, context, httpContext);
             }
-            else if (invocationResult?.Value is AspNetCoreHttpResponseData)
+            else
             {
-                // The AspNetCoreHttpResponseData implementation is
-                // simply a wrapper over the underlying HttpResponse and
-                // all APIs manipulate the request.
-                // There's no need to return this result as no additional
-                // processing is required.
-                invocationResult.Value = null;
+                HandleHttpResultInOutputBindings(context, httpContext);
             }
 
             // allows asp.net middleware to continue
             _coordinator.CompleteFunctionInvocation(invocationId);
         }
 
+        private static async void HandleHttpResult(object? httpResult, FunctionContext context, HttpContext httpContext)
```

Comment:
Please avoid `async void`.

For more information, please see https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void